src/main/kotlin/org/sert2521/infiniterecharge2020/Main.kt

```diff
@@ -1,16 +1,33 @@
 package org.sert2521.infiniterecharge2020
 
+import org.sert2521.infiniterecharge2020.autonomous.PathGenerator
 import org.sert2521.infiniterecharge2020.drivetrain.Drivetrain
+import org.sert2521.sertain.events.onEnable
+import org.sert2521.sertain.events.whileAuto
 import org.sert2521.sertain.events.whileTeleop
 import org.sert2521.sertain.robot
+import org.sert2521.sertain.subsystems.access
 import org.sert2521.sertain.subsystems.add
 
-
 suspend fun main() = robot {
     println("Robot program starting")
     add<Drivetrain>()
+    add<PathGenerator>()
+
+    onEnable {
+        val dt = access<Drivetrain>()
+        dt.gyro.reset()
+        dt.zeroEncoders()
+    }
 
     whileTeleop {
         initControls()
     }
-}
+
+    whileAuto {
+        println("Resetting gyro and encoders")
+        val dt = access<Drivetrain>()
+        dt.gyro.reset()
+        dt.zeroEncoders()
+    }
+}
```

Comment:
Add a newline here

Reply:
Do we have `ktlint` rolling on this branch yet?